isis/tests/SpiceRotationTests.cpp

```diff
@@ -0,0 +1,822 @@
+  // These are angles so we can't use vector comparison
+  vector<Angle> poleRa = ioRot.poleRaCoefs();
+  EXPECT_EQ(poleRa.size(), 3);
+  EXPECT_NEAR(poleRa[0].degrees(), 268.05, testTolerance);
+  EXPECT_NEAR(poleRa[1].degrees(), -0.009, testTolerance);
+  EXPECT_NEAR(poleRa[2].degrees(), 0.0, testTolerance);
+
+  vector<Angle> poleDec = ioRot.poleRaCoefs();
+  EXPECT_EQ(poleDec.size(), 3);
+  EXPECT_NEAR(poleDec[0].degrees(), 64.5, testTolerance);
+  EXPECT_NEAR(poleDec[1].degrees(), 0.003, testTolerance);
+  EXPECT_NEAR(poleDec[2].degrees(), 0.0, testTolerance);
+
+  vector<Angle> prMer = ioRot.poleRaCoefs();
+  EXPECT_EQ(prMer.size(), 3);
+  EXPECT_NEAR(prMer[0].degrees(), 200.39, testTolerance);
+  EXPECT_NEAR(prMer[1].degrees(), 203.48895, testTolerance);
+  EXPECT_NEAR(prMer[2].degrees(), 0.0, testTolerance);
+
+  vector<Angle> sysNutPrec0 = ioRot.sysNutPrecConstants();
+  EXPECT_EQ(sysNutPrec0.size(), 15);
+  EXPECT_NEAR(sysNutPrec0[0].degrees(), 73.32, testTolerance);
+  EXPECT_NEAR(sysNutPrec0[1].degrees(), 24.62, testTolerance);
+  EXPECT_NEAR(sysNutPrec0[2].degrees(), 283.9, testTolerance);
+  EXPECT_NEAR(sysNutPrec0[3].degrees(), 355.8, testTolerance);
+  EXPECT_NEAR(sysNutPrec0[4].degrees(), 119.9, testTolerance);
+  EXPECT_NEAR(sysNutPrec0[5].degrees(), 229.8, testTolerance);
+  EXPECT_NEAR(sysNutPrec0[6].degrees(), 352.25, testTolerance);
+  EXPECT_NEAR(sysNutPrec0[7].degrees(), 113.35, testTolerance);
+  EXPECT_NEAR(sysNutPrec0[8].degrees(), 146.64, testTolerance);
+  EXPECT_NEAR(sysNutPrec0[9].degrees(), 49.24, testTolerance);
+  EXPECT_NEAR(sysNutPrec0[10].degrees(), 99.360714, testTolerance);
+  EXPECT_NEAR(sysNutPrec0[11].degrees(), 175.89537, testTolerance);
+  EXPECT_NEAR(sysNutPrec0[12].degrees(), 300.32316, testTolerance);
+  EXPECT_NEAR(sysNutPrec0[13].degrees(), 114.0123, testTolerance);
+  EXPECT_NEAR(sysNutPrec0[14].degrees(), 49.511251, testTolerance);
+
+  vector<Angle> sysNutPrec1 = ioRot.sysNutPrecCoefs();
+  EXPECT_EQ(sysNutPrec1.size(), 15);
+  EXPECT_NEAR(sysNutPrec1[0].degrees(), 91472.9, testTolerance);
+  EXPECT_NEAR(sysNutPrec1[1].degrees(), 45137.2, testTolerance);
+  EXPECT_NEAR(sysNutPrec1[2].degrees(), 4850.7, testTolerance);
+  EXPECT_NEAR(sysNutPrec1[3].degrees(), 1191.3, testTolerance);
+  EXPECT_NEAR(sysNutPrec1[4].degrees(), 262.1, testTolerance);
+  EXPECT_NEAR(sysNutPrec1[5].degrees(), 64.3, testTolerance);
+  EXPECT_NEAR(sysNutPrec1[6].degrees(), 2382.6, testTolerance);
+  EXPECT_NEAR(sysNutPrec1[7].degrees(), 6070, testTolerance);
+  EXPECT_NEAR(sysNutPrec1[8].degrees(), 182945.8, testTolerance);
+  EXPECT_NEAR(sysNutPrec1[9].degrees(), 90274.4, testTolerance);
+  EXPECT_NEAR(sysNutPrec1[10].degrees(), 4850.4046, testTolerance);
+  EXPECT_NEAR(sysNutPrec1[11].degrees(), 1191.9605, testTolerance);
+  EXPECT_NEAR(sysNutPrec1[12].degrees(), 262.5475, testTolerance);
+  EXPECT_NEAR(sysNutPrec1[13].degrees(), 6070.2476, testTolerance);
+  EXPECT_NEAR(sysNutPrec1[14].degrees(), 64.3, testTolerance);
```

Comment:
I tested some stuff using [transform](https://en.cppreference.com/w/cpp/algorithm/transform) and [bind](https://en.cppreference.com/w/cpp/utility/functional/bind) to convert these angle vectors into double vectors, but it took almost as much space and didn't seem worth the extra complexity.

Another option I looked at was making the vector comparison templated, but the Angle class doesn't like standard arithmetic and comparison.